ui/src/progress_table_cell_type.tsx

```diff
@@ -17,19 +17,42 @@ import { F, S } from './core'
 import React from 'react'
 import { stylesheet } from 'typestyle'
 import { ProgressArc } from './parts/progress_arc'
+import { ProgressBar } from './parts/progress_bar'
 import { clas, cssVar } from './theme'
 
 const
   css = stylesheet({
-    container: {
+    barContainer: {
       position: 'relative',
+      display: 'inline-flex',
+      width: 100,
+    },
+
+    spinnerContainer: {
+      position: 'relative',
+      display: 'inline-flex',
+      width: 100,
+      height: 50
+    },
+    
+    bar: {
+      position: 'absolute',
+      top: '50%', left: 0,
+      width: 50,
+    },
+
+    spinner: {
+      position: 'absolute',
+      top: 0, left: 0, bottom: 0, right: 0,
```

Comment:
What is absolute positioning here for?

Reply:
I just reused the previous code. It positions the percentage inside the spinner. I'll organize the code a bit better to make it more clear.